google/cloud/forseti/services/inventory/base/resources.py

```diff
@@ -107,6 +106,16 @@ def parent(self):
     def key(self):
         raise NotImplementedError('Class: {}'.format(self.__class__.__name__))
 
+    def add_warning(self, warning):
+        warning_message = '{}\n'.format(warning)
```

Comment:
What about making self._warning a list that you append to, and then in get_warning() return '\n'.join(self._warning) ? Then you don't need to build a new string for each new warning.

Reply:
Fixed this one.